microsoft/testsuites/nested/nested.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import time
+from typing import Any, Dict
+
+from assertpy import assert_that
+
+from lisa import RemoteNode, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.schema import Node
+from lisa.tools import Cat, Echo, Qemu, Sshpass, Wget
+
+
+@TestSuiteMetadata(
+    area="nested",
+    category="functional",
+    description="""
+    This test suite is used to run nested vm related tests.
+    """,
+)
+class Nested(TestSuite):
+    NESTED_VM_TEST_FILE_NAME = "message.txt"
+    NESTED_VM_TEST_FILE_CONTENT = "Message from L1 vm!!"
+    NESTED_VM_TEST_PUBLIC_FILE_URL = "http://www.github.com"
+
+    def get_l2_vm(
```

Comment:
`_connect_nested_vm`? It won't be shared in other test suites, the private method is good.